### PR TITLE
Add deprecated repair JMX API tests

### PR DESCRIPTION
**Pull request by yukim**
Test for [CASSANDRA-9570](https://issues.apache.org/jira/browse/CASSANDRA-9570).
In the future, when ```@Deprecated``` repair methods are removed, this test will be obsolete.

### CODE REVIEW

File: deprecated_repair_test.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+from cassandra.query import SimpleStatement
+
+from dtest import Tester, debug
+from tools import since, insert_c1c2, query_c1c2
+from jmxutils import JolokiaAgent, make_mbean, remove_perf_disable_shared_mem
+
+@since("2.2")
+class TestDeprecatedRepairAPI(Tester):
+    """
+    Test if depreated repair JMX API runs with expected parameters
```
Comment (ptnapoleon) [inline deprecated_repair_test.py]: Can you add a line `@jira_ticket CASSANDRA-9570` in this docstring?

**Comment by ptnapoleon** [comment]
Do these deprecated methods only exist in 2.2? Why are we not testing them in previous versions?

**Comment by yukim** [comment]
These are deprecated in 2.2 and no longer used from ```nodetool repair```. So 2.1 and before are (kind of) tested through repair test.
From 2.2, ```nodetool``` uses unified ```repairAsync``` method.

**Comment by ptnapoleon** [comment]
+1, thanks.

**Pull request merged**